From 7e5740eb9bdb6229a43976c26c9692aa1106d2be Mon Sep 17 00:00:00 2001
From: Shinobu Uehara <shinobu.uehara.xc@renesas.com>
Date: Fri, 14 Jun 2013 13:24:58 +0900
Subject: [PATCH 0507/1083] ARM: shmobile: r8a7790: Fix HS-USB register access
 using ioremap

Signed-off-by: Shinobu Uehara <shinobu.uehara.xc@renesas.com>
(cherry picked from commit b0c0e99f550b1dffd43fca0ef35145791515edd7)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 arch/arm/mach-shmobile/setup-r8a7790.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-shmobile/setup-r8a7790.c b/arch/arm/mach-shmobile/setup-r8a7790.c
index 7c1eabc..9856e24 100644
--- a/arch/arm/mach-shmobile/setup-r8a7790.c
+++ b/arch/arm/mach-shmobile/setup-r8a7790.c
@@ -534,7 +534,7 @@ static void __init usbh_pci_int_enable(int ch)
 static int __init usbh_init(void)
 {
 	struct clk *clk_hs, *clk_ehci;
-	u32 hs_usb = 0xE6590000;
+	void __iomem *hs_usb = ioremap_nocache(0xE6590000, 0x1ff);
 	unsigned int ch;
 
 	clk_hs = clk_get(NULL, "hs_usb");
@@ -550,7 +550,7 @@ static int __init usbh_init(void)
 	clk_enable(clk_ehci);
 
 	/* Set EHCI for UGCTRL2 */
-	__raw_writel(0x00000011, (hs_usb + 0x184));
+	iowrite32(0x00000011, (hs_usb + 0x184));
 
 	for (ch = 0; ch < SHUSBH_MAX_CH; ch++) {
 		/* internal pci-bus bridge initialize */
@@ -565,6 +565,7 @@ static int __init usbh_init(void)
 		/* pci int enable */
 		usbh_pci_int_enable(ch);
 	}
+	iounmap(hs_usb);
 
 	return 0;
 }
-- 
1.8.3.2

